### DISCUSSION TITLE
Import templates from another YAML file

### DISCUSSION THREAD

**Discussion by francisco1c** [argoproj/argo-cd#9916] [Ideas]
Hello,

Recently I've been using argo to implement a data pipeline that is quite complex, and I've been struggling to handle modularity in a direct and easy way. The main reason for that is that I can not have my different templates stored in different files, so I end up with a workflow yaml file that is simply too big.

I know that I can use workflow templates with the templateRef for that, but those first need to be deployed to the cluster. In a scenario that there is a lot of developers working in parallel, managing those deployed workflows becomes a little annoying, and also gives me the headache of dealing with versioning of the different workflow templates and more deployment processes overhead.

Wouldn't it be super good to allow templates to be structured as a proper code language, and be imported or referenced locally, when the submit command is submitted?

**Comment by crenshaw-dev** [comment]
I think you probably want this Discussions board: https://github.com/argoproj/argo-workflows/discussions

I know there is a Python library for writing more complex Workflows. You could also generate the Workflows (or WorkflowTemplates) using Helm. But I think you'll find more help on the other repo.  :-)

**Reply by francisco1c** [reply]
I will post on that board, thanks for the suggestions.
